src/Console/Commands/JsonApiGeneratorCommand.php

```diff
@@ -0,0 +1,250 @@
+<?php
+
+namespace CloudCreativity\LaravelJsonApi\Console\Commands;
+
+use Illuminate\Console\GeneratorCommand as LaravelGeneratorCommand;
+use Illuminate\Filesystem\Filesystem;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputArgument;
+
+abstract class JsonApiGeneratorCommand extends LaravelGeneratorCommand
+{
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $name;
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description;
+
+    /**
+     * The type of class being generated.
+     *
+     * @var string
+     */
+    protected $type;
+
+    /**
+     * Whether the resource type is non-dependent on eloquent.
+     *
+     * @var boolean
+     */
+    protected $isIndependent = false;
+
+    /**
+     * Whether the resource should use eloquent implementations.
+     *
+     * @var boolean
+     */
+    protected $useEloquent;
+
+    /**
+     * The folder within the root namespace, where files should be generated.
+     *
+     * @var string
+     */
+    protected $subNamespace;
+
+    /**
+     * Whether generated files should be grouped by their resource files.
+     *
+     * @var mixed
+     */
+    protected $namespaceByResource;
+
+    /**
+     * The location of all generator stubs
+     *
+     * @var string
+     */
+    private $stubsDirectory = __DIR__.'/../../../stubs';
+
+    /**
+     * Create a new config clear command instance.
+     *
+     * @return void
+     */
+    public function __construct(Filesystem $files)
+    {
+        parent::__construct($files);
+
+        $this->useEloquent = config('json-api.generator.use-eloquent');
+        $this->subNamespace = config('json-api.generator.namespace');
+        $this->namespaceByResource = config('json-api.generator.by-resource');
+    }
+
+    /**
+     * Build the class with the given name.
+     * Remove the base controller import if we are already in base namespace.
+     *
+     * @param  string  $name
+     * @return string
+     */
+    protected function buildClass($name)
+    {
+        $namespace = $this->getNamespace($name);
+
+        $stub = $this->files->get($this->getStub());
+
+        $this->replaceNamespace($stub, $name)
+                ->replaceResourceType($stub, $this->getResourceName());
+
+        return $stub;
+    }
+
+    /**
+     * Replace the value of the resource type constant
+     *
+     * @param mixed $stub
+     * @param mixed $resource
+     */
+    protected function replaceResourceType(&$stub, $resource)
+    {
+        $stub = str_replace(
+            'dummyResourceType', camel_case($resource), $stub
```

Comment:
Change to: `snake_case($resource, '-')`

Because if generating a multi-word resource type, the JSON API convention is for it to be a dasherized string